src/openrct2/scripting/ScPark.hpp

```diff
@@ -24,6 +25,209 @@
 
 namespace OpenRCT2::Scripting
 {
+    static constexpr const char* ParkMessageTypeStrings[] = { "attraction", "peep_on_attraction", "peep",   "money",
+                                                              "blank",      "research",           "guests", "award",
+                                                              "chart" };
+
+    inline uint8_t GetParkMessageType(const std::string& key)
+    {
+        uint8_t i = 0;
+        for (const auto& k : ParkMessageTypeStrings)
+        {
+            if (k == key)
+            {
+                return NEWS_ITEM_RIDE + i;
+            }
+            i++;
+        }
+        return NEWS_ITEM_BLANK;
```

Comment:
What about this (or similar?)
```suggestion
        auto it = std::find(ParkMessageTypeStrings.begin(), ParkMessageTypeStrings.end(), key);
        return it != ParkMessageTypeStrings.end() ? std::distance(ParkMessageTypeStrings.begin(), it) + NEWS_ITEM_RIDE : NEWS_ITEM_BLANK;
```